ObjectOrientedProgramming/ClassesAndObjects/SceneTransition/src/scenes.cpp

```diff
@@ -0,0 +1,25 @@
+#include "scenes.hpp"
+
+bool SceneManager::initialize() {
+    dynamicScene.activate();
+    return true;
+}
+
+Scene* SceneManager::getCurrentScene() {
+    return &dynamicScene;
+}
+
+Scene* SceneManager::transitionScene(SceneID sceneID, Scene* currentScene, unsigned int score) {
+    switch (sceneID) {
+        case SceneID::DYNAMIC_GAME_FIELD:
+            dynamicScene.activate();
+            return &dynamicScene;
+        case SceneID::LEADERBOARD:
+            leaderboardScene.activate();
+            leaderboardScene.recieveScore(score);
+            dynamicScene.deactivate();
+            return &leaderboardScene;
```

Comment:
Let's implement a refactoring here. 

1. Let's rollback `transitionScene` signature to previous version, so it takes only `SceneID` argument. 
2. Let's add `Scene* currentScene` field to `SceneManager`. You should also update it when the scene transitions occur.
3. Here, when a transition to `SceneID::LEADERBOARD` is requested, you try to case `currentScene` to `GameplayDynamicScene`, if the cast is successfull, you obtain the current score, and then set it to `leaderboardScene`.